tests/test_parametric_components/test_PortCutterCircular.py

```diff
@@ -1,19 +1,57 @@
 
+import math
 import unittest
 
 import paramak
 
+import pytest
 
-# class test_component(unittest.TestCase):
-# TODO: fix issue 548
-#     def test_creation(self):
-#         """Checks a PortCutterCircular creation."""
 
-#         test_component = paramak.PortCutterCircular(
-#             distance=3,
-#             z_pos=0.25,
-#             radius=0.1,
-#             azimuth_placement_angle=[0, 45, 90, 180]
-#         )
+class test_component(unittest.TestCase):
 
-#         assert test_component.solid is not None
+    def setUp(self):
+        self.test_shape = paramak.PortCutterCircular(
+            distance=300, radius=20
+        )
+
+    def test_default_parameters(self):
+        """Checks that the default parameters of a PortCutterCircular are correct."""
+
+        assert self.test_shape.center_point == (0, 0)
+        assert self.test_shape.workplane == "ZY"
+        assert self.test_shape.rotation_axis == "Z"
+        assert self.test_shape.extrusion_start_offset == 1
+        assert self.test_shape.stp_filename == "PortCutterCircular.stp"
+        assert self.test_shape.stl_filename == "PortCutterCircular.stl"
+        assert self.test_shape.name == "circular_port_cutter"
+        assert self.test_shape.material_tag == "circular_port_cutter_mat"
+
+    def test_creation(self):
+        """Creates a circular port cutter using the PortCutterCircular parametric
+        component and checks that a cadquery solid is created."""
+
+        assert self.test_shape.solid is not None
+        assert self.test_shape.volume > 1000
+
+    def test_relative_volume(self):
+        """Creates PortCutterCircular shapes and checks that their relative volumes
+        are correct."""
+
+        test_volume = self.test_shape.volume
+
+        self.test_shape.extrusion_start_offset = 20
+        self.test_shape.azimuth_placement_angle = [0, 90, 180, 270]
+
+        assert self.test_shape.volume == pytest.approx(test_volume * 4)
+
+    def test_absolute_volume(self):
+        """Creates a PortCutterCircular shape and checks that its volume is correct."""
+
+        assert self.test_shape.volume == pytest.approx(math.pi * (20**2) * 300)
```

Comment:
Careful here, if another test is added inbetween and changes the instance attributes, this test will fail. To be safe, you can add `self.test_shape.radius= 20`, `self.test_shape.distance = 300` and  `self.test_shape.extrusion_start_offset = 0.0` to ensure that the correct shape is made.

Reply:
In the current setup, any changes to the test_shape made in a test are unique to that particular test - they do not propagate to other tests. I.e. each test imports the shape defined in setUp(), so adding a new test which changes the shape attributes before this test will have no effect.

Reply:
Ah ok super then. I thought the issue we had a while ago where stuff was changed but I think that was inside a test wasn't it ? Nevermind 😄